compiler/src/iree/compiler/Reducer/Strategies/ReduceLinalgOnTensorsDelta.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <random>
+
+#include "iree/compiler/Reducer/Strategies/DeltaStrategies.h"
+
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/Passes.h"
+
+using namespace mlir;
+using namespace mlir::iree_compiler;
+
+void mlir::iree_compiler::reduceLinalgOnTensorsDelta(ChunkManager &chunker,
+                                                     WorkItem &workItem) {
+  ModuleOp module = workItem.getModule();
+
+  SmallVector<linalg::LinalgOp> linalgOps;
+  module.walk([&](linalg::LinalgOp op) {
+    if (!op.hasTensorSemantics())
+      return;
+    // Op should have at least on tensor input, otherwise it is not worth
```

Comment:
Could you expand on this? Why isn't it worth it?

Reply:
If there is no tensor input, then the operation is similar to a fill operation. This pass is supposed to reduce the number of linalg operations, not reduce the complexity of the linalg operation itself. Added inline documentation about this.

Reply:
On second thought, reducing might also help as it reduces it to an "identity" fill. Added a TODO about this.